File: webapps/smsc/WEB-INF/src/ru/novosoft/smsc/admin/AdminException.java
/*
 * Created by igork
 * Date: Feb 28, 2002
 * Time: 5:08:17 PM
 */
package ru.novosoft.smsc.admin;


public class AdminException extends Exception
{
	public AdminException(String s)
	{
		super(s);
	}

  public AdminException(String s, Throwable cause)
	{
		super(s, cause);
	}
}

File: src/admin/AdminException.cpp
#include "AdminException.h"

#include <string.h>

namespace smsc {
namespace admin {

AdminException::AdminException(const char *	const message)
	throw ()
	: std::exception()
{
	msg = new char[strlen(message)+1];
	strcpy(msg, message);
}

AdminException::AdminException(const AdminException &a)
	throw ()
	: std::exception()
{
	msg = new char[strlen(a.msg)+1];
	strcpy(msg, a.msg);
}

/*AdminException::AdminException(AdminException a) {
	msg = new char[strlen(a.msg)+1];
	strcpy(msg, a.msg);
}*/

AdminException::~AdminException()
	throw ()
{
	delete[] msg;
}

}
}

